File: DSAlgorithms/Leetcode/c++/arrays_hash/169-majority_element.cpp
int majorityElement(vector<int>& nums) {
    int count = 0;
    int majority_element = 0;
    for(int i =0; i < nums.size(); i++)
    {
        if(count == 0)
            majority_element = nums[i];
        if(nums[i] == majority_element)
            count++;
        else count--;
    }
    return majority_element;
}

File: dsa_algo/leetcode/ts/src/arrays_hash/169-majority_element.ts
const  majorityElement = (nums: number[]): number => {
    let count = 0;
    let majority_element = 0;
    for(let i = 0; i < nums.length; i++)
    {
        if(count === 0)
            majority_element = nums[i];
        if(majority_element === nums[i])
            count++;
        else 
            count--;
    }
    return majority_element;
    
}
